Added basic communication with PokeApi

kamil-matula opened the pull request:
Added methods responsible for gathering list of all pokemons, details of specific pokemon and creating custom JSON objects. Added some hooks and simple view to test these new features.

kamil-matula committed:
Added basic communication with PokeApi

Added methods responsible for gathering list of all pokemons, details of specific pokemon and creating custom JSON objects. Added some hooks and simple view to test these new features.

Blazevarjo commented on App.js:
```diff
@@ -1,12 +1,29 @@
+  const [isLoading, setIsLoading] = useState(true);
+  const [dataToDisplay, setDataToDisplay] = useState("");
+  useEffect(() => {
+    pokemonByNameOrNumber(4, setDataToDisplay, setIsLoading, false); 
```
I think it's better to have this function to return data, rather than passing state. For example:
```suggestion
    setIsLoading(true);
    pokemonByNameOrNumber(4)
    .then( pokemon => {
        setDataToDisplay(pokemon)
        setIsLoading(false);
    }); 
```

kamil-matula committed:
Refactored API methods

Removed arguments from methods which are responsible for communication with PokeAPI. The functions now return data instead of passing the state.
